internal/otelmatchers/trace_matchers.go

```diff
@@ -0,0 +1,116 @@
+package otelmatchers
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+
+	"github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+func ConsistOfSpansWithIDs(expectedSpanIDs []pcommon.SpanID) types.GomegaMatcher {
+	return gomega.WithTransform(func(actual interface{}) ([]pcommon.SpanID, error) {
+		actualBytes, ok := actual.([]byte)
+		if !ok {
+			return nil, fmt.Errorf("ConsistOfSpansWithIDs rqquires a []byte, but got %T", actual)
+		}
+
+		actualTraces, err := unmarshalOTLPJSONTraces(actualBytes)
+		if err != nil {
+			return nil, fmt.Errorf("ConsistOfSpansWithIDs requires a valid OTLP JSON document: %v", err)
+		}
+
+		actualSpans := getAllSpans(actualTraces)
+
+		var actualSpanIDs []pcommon.SpanID
+		for _, span := range actualSpans {
+			actualSpanIDs = append(actualSpanIDs, span.SpanID())
+		}
+		return actualSpanIDs, nil
+	}, gomega.ConsistOf(expectedSpanIDs))
+}
+
+func EachHaveTraceID(expectedTraceID pcommon.TraceID) types.GomegaMatcher {
```

Comment:
The name for the `EachHaveTraceID` function is too vague to me. As it is called for an HTTPBody, it is unclear what `Each...` implies. I suggest aligning it with the `ConsistOfSpansWithIDs` one.

Reply:
I don't like the names, to be honest. Changed them, please check.